modules/apps/account/account-test/src/testIntegration/java/com/liferay/account/admin/web/internal/portlet/action/test/InviteAccountUsersMVCActionCommandTest.java

```diff
@@ -0,0 +1,175 @@
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.test.rule.PermissionCheckerMethodTestRule;
+
+import java.util.List;
+
+import javax.portlet.ActionRequest;
+import javax.portlet.ActionResponse;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Matyas Wollner
+ */
+@RunWith(Arquillian.class)
+@Sync
+public class InviteAccountUsersMVCActionCommandTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@Before
+	public void setUp() throws Exception {
+		_accountEntry = AccountEntryTestUtil.addAccountEntry();
+
+		_group = GroupTestUtil.addGroup();
+
+		_company = _companyLocalService.getCompany(_group.getCompanyId());
+
+		_layout = LayoutTestUtil.addTypeContentLayout(_group);
+
+		_serviceContext = ServiceContextTestUtil.getServiceContext(
+			_group.getGroupId(), TestPropsValues.getUserId());
+
+		_serviceContext.setCompanyId(TestPropsValues.getCompanyId());
+
+		ServiceContextThreadLocal.pushServiceContext(_serviceContext);
+	}
+
+	@Test
+	public void testAddExistingAndNonexistingEmailAddresses() throws Exception {
+		MockLiferayPortletActionRequest mockLiferayPortletActionRequest =
+			_getMockLiferayPortletActionRequest();
+
+		User user = _userLocalService.fetchUserByEmailAddress(
+			_company.getCompanyId(), "test@liferay.com");
+
+		mockLiferayPortletActionRequest.addParameter("count", "1");
+		mockLiferayPortletActionRequest.addParameter(
+			"emailAddresses0",
+			new String[] {"test@liferay.com", "buyer@liferay.com"});
+
+		ReflectionTestUtil.invoke(
+			_mvcActionCommand, "doTransactionalCommand",
+			new Class<?>[] {ActionRequest.class, ActionResponse.class},
+			mockLiferayPortletActionRequest,
+			new MockLiferayPortletActionResponse());
+
+		List<User> accountEntryUsers = _accountEntry.fetchUsers();
+
+		Assert.assertFalse(accountEntryUsers.contains(user));
+
+		List<Ticket> tickets = _ticketLocalService.getTickets(
+			AccountEntry.class.getName(), _accountEntry.getAccountEntryId(),
+			AccountTicketConstants.TYPE_USER_INVITATION);
+
+		Assert.assertFalse(tickets.isEmpty());
+
+		for (Ticket ticket : tickets) {
+			String extraInfo = ticket.getExtraInfo();
+
+			Assert.assertTrue(extraInfo.contains("buyer@liferay.com"));
+		}
+	}
+
+	private MockLiferayPortletActionRequest
+			_getMockLiferayPortletActionRequest()
+		throws Exception {
+
+		MockLiferayPortletActionRequest mockLiferayPortletActionRequest =
+			new MockLiferayPortletActionRequest();
+
+		mockLiferayPortletActionRequest.setAttribute(
+			WebKeys.THEME_DISPLAY, _getThemeDisplay());
+
+		mockLiferayPortletActionRequest.addParameter(
+			"accountEntryId",
+			String.valueOf(_accountEntry.getAccountEntryId()));
```

Comment:
Can you move this in the test instead of having it in a dedicated method, so we can have all the addParameter in the same place